rpc/RPCTransaction.java

```diff
@@ -127,10 +127,16 @@ public void rollback() {
 
     @Override
     public void close() {
-        close(new Response.Done.Completed());
+        closeFromClient(new Response.Done.Completed());
     }
 
-    private void close(Response.Done doneResponse) {
+    private void closeFromServer(Response.Done doneResponse) {
```

Comment:
Hmm. I think a better name for this method would be `onServerClose` - it's something that gets triggered remotely, not by the client. In which case we can take out `closeFromClient` and simply move its code into `close`.